src/bindgen/library.rs

```diff
@@ -443,4 +450,53 @@ impl Library {
             x.mangle_paths(&monomorphs);
         }
     }
+
+    // Apply associated type replacement to everything
+    fn replace_assoc_types(&mut self){
+
+        let assoc_map = &self.assoc_types;
+
+        // Replace associated types in function signatures
+        for func in self.functions.iter_mut(){
+            // Check return type
+            if let Some(ty) = find_concrete(&func.ret, assoc_map){
+                func.ret = ty
+            };
+
+            // Check arguments type
+            for arg in func.args.iter_mut(){
+                if let Some(ty) = find_concrete(&arg.ty, assoc_map){
+                    arg.ty = ty
+                };
+            }
+        }
+
+        // Replace associated types in structs
+        self.structs.for_all_items_mut(|struct_| {
+            for field in struct_.fields.iter_mut(){
+                if let Some(ty) = find_concrete(&field.ty, assoc_map){
+                    field.ty = ty;
+                }
+            }
+        });
+
+        // Replace associated types in constants
+        self.constants.for_all_items_mut(|const_|{
+            if let Some(ty) = find_concrete(&const_.ty, assoc_map){
+                const_.ty = ty;
+            }
+        });
+    
+    }
+
+}
+
+// Search and replace specific associated types with concrete types
+fn find_concrete(ty: &Type, type_map: &HashMap<AssocTypeId, Type>) -> Option<Type> {
+    if let Type::Path(GenericPath{assoc: Some(id), ..}) = &ty{
+        if let Some(ty) = type_map.get(id){
+            return Some(ty.clone());
+        }
+    }
+    None
```

Comment:
```suggestion
fn find_concrete(ty: &Type, type_map: &HashMap<AssocTypeId, Type>) -> Option<&Type> {
    if let Type::Path(GenericPath{assoc: Some(id), ..}) = &ty{
        if let Some(ty) = type_map.get(id){
            return Some(ty);
        }
    }
    None
```

it's better to clone return value on call site